kerastuner/engine/tuner.py

```diff
@@ -145,8 +145,14 @@ def run_trial(self, trial, *fit_args, **fit_kwargs):
         copied_fit_kwargs['callbacks'] = callbacks
 
         model = self.hypermodel.build(trial.hyperparameters)
+        self.on_fit_begin(model, trial.hyperparameters, *fit_args, **copied_fit_kwargs)
         model.fit(*fit_args, **copied_fit_kwargs)
 
+    def on_fit_begin(model, hp, *fit_args, **fit_kwargs):
```

Comment:
We may want to call it `on_train_begin` for consistency with what it's called in callbacks.

Reply:
Do we want to commit to supporting a hook here, or should we ask autokeras to override `run_trial` to support this?

I'm worried that it will be somewhat confusing when to supply a Callback with `on_train_begin` versus when to subclass `Tuner`, and we will probably have a native solution for preprocessing layers soon

Reply:
Discussed offline, a private hook is good for now